os/board/rtl8721csm/src/component/common/bluetooth/realtek/sdk/example/ble_central/ble_tizenrt_central_client_app.c

```diff
@@ -583,8 +585,9 @@ void ble_tizenrt_central_app_handle_conn_state_evt(uint8_t conn_id, T_GAP_CONN_S
                 if (g_master_link_num){
                     g_master_link_num --;
                 }
-            uint32_t connid = (uint32_t) conn_id;
-            if(ble_tizenrt_central_send_callback_msg(BLE_TIZENRT_DISCONNECTED_MSG, (void *) connid) == false)
+            uint32_t temp = (uint32_t) conn_id;
+            temp = (temp << 16) | (disc_cause & 0xff);
```

Comment:
Please add a comment about this logic. Also, can we have a #define which can perform this action?

Reply:
'ble msg handler' can send only 1 pointer as a parameter. So to send 2 params, I combined them by bit shifting.
The total of 'conn_id' and 'disc_cause' is 24 bits, so I used 32 bits variable.
The first 16-bits are for conn_id, rest of bits are for disc_cause.